cli/src/stake.rs

```diff
@@ -515,37 +551,58 @@ pub fn parse_stake_authorize(
     matches: &ArgMatches<'_>,
     default_signer_path: &str,
     wallet_manager: Option<&Arc<RemoteWalletManager>>,
-    stake_authorize: StakeAuthorize,
+    stake_authorizations: &[StakeAuthorize],
 ) -> Result<CliCommandInfo, CliError> {
     let stake_account_pubkey =
         pubkey_of_signer(matches, "stake_account_pubkey", wallet_manager)?.unwrap();
-    let new_authorized_pubkey =
-        pubkey_of_signer(matches, "authorized_pubkey", wallet_manager)?.unwrap();
-    let authority_flag = match stake_authorize {
-        StakeAuthorize::Staker => STAKE_AUTHORITY_ARG.name,
-        StakeAuthorize::Withdrawer => WITHDRAW_AUTHORITY_ARG.name,
-    };
+    let mut new_authorizations = Vec::new();
+    let mut bulk_signers = Vec::new();
+    for stake_authorize in stake_authorizations {
+        let (authority_flag, new_authority_flag) = match stake_authorize {
+            StakeAuthorize::Staker => (STAKE_AUTHORITY_ARG.name, "new_stake_authority"),
+            StakeAuthorize::Withdrawer => (WITHDRAW_AUTHORITY_ARG.name, "new_withdraw_authority"),
+        };
+        if let Some(new_authority_pubkey) =
+            pubkey_of_signer(matches, new_authority_flag, wallet_manager)?
+        {
+            let (authority, authority_pubkey) = signer_of(matches, authority_flag, wallet_manager)?;
+            new_authorizations.push((*stake_authorize, new_authority_pubkey, authority_pubkey));
+            bulk_signers.push(authority);
+        } else if stake_authorizations.len() == 1 {
```

Comment:
Since you are using `required_unless` on the authorization args, I think this case is unreachable. How were you able to make `stake_authorizations.len() == 1`?

Reply:
All three of the `stake-authorize*` subcommands follow this code path.

https://github.com/solana-labs/solana/blob/26d9a03f53c6434ade7e2580257a38939dc446ef/cli/src/cli.rs#L649-L666

This case covers the `unwrap()` that was formerly at L523.  It _should_ only be tagged by a malformed test

Reply:
Actually, can't you still use `unwrap()` instead of `if let` on L566-567? That would be much easier on the brain than this else case.

Reply:
That would get hit by `stake-authorize` being used in the single-authority
cases.

We should be deprecating the individual authority subcommands, IMO